webapp/channels/src/components/advanced_text_editor/send_button/send_post_options/core_menu_options.tsx

```diff
@@ -23,6 +30,69 @@ function CoreMenuOptions({handleOnSelect, channelId}: Props) {
         isDM,
     } = useTimePostBoxIndicator(channelId);
 
+    const recentlyUsedCustomDate = useSelector((state: GlobalState) => getPreference(state, scheduledPosts.SCHEDULED_POSTS, scheduledPosts.RECENTLY_USED_CUSTOM_TIME));
+
+    interface RecentlyUsedCustomDate {
+        update_at?: number;
+        timestamp?: number;
+    }
+
+    let recentlyUsedCustomDateVal: RecentlyUsedCustomDate = {};
+
+    if (recentlyUsedCustomDate) {
+        try {
+            recentlyUsedCustomDateVal = JSON.parse(recentlyUsedCustomDate) as RecentlyUsedCustomDate;
+        } catch (e) {
+            recentlyUsedCustomDateVal = {};
+        }
+    }
+
+    function isTimestampWithinLast30Days(timestamp: number, timeZone = 'UTC') {
+        if (!timestamp || isNaN(timestamp)) {
+            return false;
+        }
+        const usedDate = DateTime.fromMillis(timestamp).setZone(timeZone);
+        const now = DateTime.now().setZone(timeZone);
+        const thirtyDaysAgo = now.minus({days: 30});
+
+        return usedDate >= thirtyDaysAgo && usedDate <= now;
+    }
+
+    let recentCustomTime = null;
+    const handleRecentlyUsedCustomTime = useCallback((e) => handleOnSelect(e, recentlyUsedCustomDateVal.timestamp!), [handleOnSelect, recentlyUsedCustomDateVal.timestamp]);
+    if (
+        recentlyUsedCustomDateVal &&
+        typeof recentlyUsedCustomDateVal.update_at === 'number' &&
+        typeof recentlyUsedCustomDateVal.timestamp === 'number' &&
+        isTimestampWithinLast30Days(recentlyUsedCustomDateVal.update_at, userCurrentTimezone)
+    ) {
+        const timestamp = (
+            <Timestamp
+                value={recentlyUsedCustomDateVal.timestamp}
+                timeZone={userCurrentTimezone}
+                useRelative={false}
+                useDate={{weekday: 'long'}}
+                useTime={{hour: 'numeric', minute: 'numeric'}}
+            />
+        );
+
+        recentCustomTime = [
+            <Menu.Separator key='recent_custom_separator'/>,
+            <Menu.Item
+                key='recently_used_custom_time'
+                onClick={handleRecentlyUsedCustomTime}
+                labels={timestamp}
+                className='core-menu-options'
+                trailingElements={(
+                    <FormattedMessage
+                        id='create_post_button.option.schedule_message.options.recently_used_custom_time'
+                        defaultMessage='Recently used custom time'
+                    />
+                )}
```

Comment:
There is probably not much to gain, but for "future proofing" I like the components to be as "stable" as possible. That means that they don't pass down new props if nothing has changed.

In this particular case, when passing a JSX element to a prop, on every render we are passing a new object. So... In order to make this a bit more stable, we can move this formatted message outside of the component, to something just like:

```
const recentlyUsedCustomTimeMessage = (
    <FormattedMessage
        id='create_post_button.option.schedule_message.options.recently_used_custom_time'
        defaultMessage='Recently used custom time'
    />
);
```

Reply:
done